src/app/spans.rs

```diff
@@ -0,0 +1,107 @@
+use crate::trace_context;
+use linkerd2_error::Error;
+use opencensus_proto::trace::v1 as oc;
+use std::collections::HashMap;
+use std::{error, fmt};
+use tokio::sync::mpsc;
+
+const SPAN_KIND_SERVER: i32 = 1;
+const SPAN_KIND_CLIENT: i32 = 2;
```

Comment:
Nit/TIOLI: We could also represent this as
```rust
#[repr(i32)]
enum SpanKind {
    Server = 1,
    Client = 2,
}
```
and then use `as` to convert it into an i32 when it's used to construct the `oc::Span` below. (see https://doc.rust-lang.org/rust-by-example/custom_types/enum/c_like.html#c-like)